client/signup/steps/plans/index.jsx

```diff
@@ -30,12 +30,20 @@ export class PlansStep extends Component {
 	state = {
 		isDesktop: isDesktop(),
 		experiment: null,
+		mobileExperiment: null,
 	};
 
 	componentWillMount() {
 		loadExperimentAssignment( 'disabled_monthly_personal_premium_v2' ).then( ( experimentName ) => {
 			this.setState( { experiment: experimentName } );
 		} );
+		if ( isMobile() ) {
```

Comment:
As per Abacus, the exposure event for this experiment has flow name set to `onboarding`. However, this line of code will assign users in any signup flow that has the plans step and show the the `treatment` version to 50% of the users. For example, users in the `launch-site` flow will also see the `treatment` experience. This will not affect the outcome of the experiment, but will mean that we show `treatment` experience to some users without intending to do so.

AFAIK exposure event only determines who is counted into the metrics calculation, and does not have an effect on assignment(cc @Automattic/experimentation-platform  to correct me if I'm wrong), so we'll have to add a check in the code for flow name. 

 ```suggestion
		if ( isMobile() && 'onboarding' === this.props.flowName ) {
```